airbyte-cdk/bulk/toolkits/load-object-storage/src/test/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactoryUTest.kt

```diff
@@ -113,4 +118,77 @@ class ObjectStoragePathFactoryUTest {
+        fileNameTemplate: String,
+        remoteFileKey: String,
+        expectedPartNumber: Long,
+    ) {
+        every { pathConfigProvider.objectStoragePathConfiguration } returns
+            ObjectStoragePathConfiguration(
+                bucketPathTemplate,
+                null,
+                filePathTemplate,
+                fileNameTemplate,
+                false,
+            )
+        val stream = mockk<DestinationStream>()
+        every { stream.descriptor } returns DestinationStream.Descriptor("namespace1", "stream_abc")
+        val factory = ObjectStoragePathFactory(pathConfigProvider, null, null, timeProvider)
+
+        val matcher = factory.getPathMatcher(stream, OPTIONAL_ORDINAL_SUFFIX_PATTERN)
+
+        val result = matcher.match(remoteFileKey)
+
+        assertNotNull(result)
+        assertEquals(expectedPartNumber, result!!.partNumber)
+    }
+
+    companion object {
+        @JvmStatic
+        private fun pathTemplateMatrix(): Stream<Arguments> {
+            return Stream.of(
+                Arguments.arguments(
+                    "\${NAMESPACE}/\${STREAM_NAME}",
+                    "\${YEAR}/\${MONTH}/\${DAY}/\${NAMESPACE}_\${BIRD_DOG}_\${STREAM_NAME}_\${YEAR}_\${MONTH}_\${DAY}_\${EPOCH}_",
+                    "{part_number}{format_extension}",
+                    "namespace1/stream_abc/2024/08/30/namespace1_BIRD_DOG_stream_abc_2024_08_30_1736900845782_1",
+                    1L,
+                ),
+                Arguments.arguments(
+                    "\${NAMESPACE}/\${STREAM_NAME}",
+                    "\${YEAR}/\${MONTH}/\${DAY}/\${NAMESPACE}_\${STREAM_NAME}_\${YEAR}_\${MONTH}_\${DAY}_\${EPOCH}_",
+                    "{part_number}{format_extension}",
+                    "namespace1/stream_abc/2024/08/30/namespace1_stream_abc_2024_08_30_1736900845782_5",
+                    5L,
+                ),
+                Arguments.arguments(
+                    "\${NAMESPACE}/\${STREAM_NAME}",
+                    "\${YEAR}/\${MONTH}/\${DAY}/\${NAMESPACE}_\${STREAM_NAME}_\${YEAR}_\${MONTH}_\${DAY}_\${EPOCH}_",
+                    "{part_number}{format_extension}",
+                    "namespace1/stream_abc/2024/08/30/namespace1_stream_abc_2024_08_30_1736900845782_7",
+                    7L,
+                ),
+                Arguments.arguments(
+                    "\${NAMESPACE}/\${STREAM_NAME}/\${STREAM_NAME}/\${STREAM_NAME}/\${STREAM_NAME}",
+                    "_",
+                    "{part_number}{format_extension}",
+                    "namespace1/stream_abc/stream_abc/stream_abc/stream_abc/_0",
+                    0,
+                ),
+                Arguments.arguments(
+                    "\${NAMESPACE}/\${STREAM_NAME}/\${STREAM_NAME}/\${STREAM_NAME}/\${STREAM_NAME}",
+                    "",
+                    "{part_number}{format_extension}",
+                    "namespace1/stream_abc/stream_abc/stream_abc/stream_abc0",
+                    0,
+                ),
```

Comment:
TODO: I should probably add cases without namespace and parameterize both name/namespace